lib/coxir/struct/guild.ex

```diff
@@ -57,6 +57,156 @@ defmodule Coxir.Struct.Guild do
     |> Gateway.get
   end
 
+  @doc """
+  Modifies the name of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_name(guild, String.t) :: map
+
+  def set_name(%{id: id}, name),
+    do: set_name(id, name)
+
+  def set_name(guild, name) do
+    edit(guild, %{name: name})
+  end
+
+  @doc """
+  Modifies the region of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_region(guild, String.t) :: map
+
+  def set_region(%{id: id}, region),
+    do: set_region(id, region)
+
+  def set_region(guild, region) do
+    edit(guild, %{region: region})
+  end
+
+  @doc """
+  Modifies the icon of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_region(guild, String.t) :: map
+
+  def set_icon(%{id: id}, icon),
+    do: set_icon(id, icon)
+
+  def set_icon(guild, icon) do
+    edit(guild, %{region: icon})
+  end
+
+  @doc """
+  Modifies the splash of a given guild (VIP only).
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_splash(guild, String.t) :: map
+
+  def set_splash(%{id: id}, splash),
+    do: set_splash(id, splash)
+
+  def set_splash(guild, splash) do
+    edit(guild, %{splash: splash})
+  end
+
+  @doc """
+  Modifies the afk timeout of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_afk_timeout(guild, Integer.t) :: map
+
+  def set_afk_timeout(%{id: id}, timeout),
+    do: set_afk_timeout(id, timeout)
+
+  def set_afk_timeout(guild, timeout) do
+    edit(guild, %{afk_timeout: timeout})
+  end
+
+  @doc """
+  Modifies the id for afk channel of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_afk_channel(guild, String.t) :: map
+
+  def set_afk_channel(%{id: id}, channel_id),
+    do: set_afk_channel(id, channel_id)
+
+  def set_afk_channel(guild, channel_id) do
+    edit(guild, %{afk_channel_id: channel_id})
+  end
+
+  @doc """
+  Modifies the id of the channel to which system messages are sent of a given guild.
+
+  Returns a guild object upon success
+  or a map containing error information.
+  """
+  @spec set_system_channel(guild, String.t) :: map
+
+  def set_system_channel(%{id: id}, channel_id), #Maybe Channel module should have a shortcut to this function?
```

Comment:
What do you mean here exactly?

Reply:
While writing this I was thinking on maybe having some function like `Channel.set_system` that would directly call this function written in guild. But as it's commented it is just an idea that can be removed.